onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -347,8 +361,44 @@ def compute_range(self):
         if not self.collector:
             raise ValueError("No collector created and can't generate calibration data.")
 
-        return self.collector.get_optimal_collection_result()
+        return self.collector.compute_collection_result()
 
+class EntropyCalibrater(HistogramCalibrater):
+    def __init__(self,
+                 model,
+                 op_types_to_calibrate=[],
+                 augmented_model_path='augmented_model.onnx',
+                 method='entropy',
+                 num_quantized_bins=128):
+        '''
+        :param model: ONNX model to calibrate. It can be a ModelProto or a model path
+        :param op_types_to_calibrate: operator types to calibrate. By default, calibrate all the float32/float16 tensors.
+        :param augmented_model_path: save augmented model to this path.
+        :param method: A string. One of ['entropy', 'percentile'].
+        :param num_quantized_bins: number of quantized bins. Default 128.
+        '''
+        super(EntropyCalibrater, self).__init__(model, op_types_to_calibrate, augmented_model_path,
+                                                method=method, num_quantized_bins=num_quantized_bins)
+
+class PercentileCalibrater(HistogramCalibrater):
+    def __init__(self,
+                 model,
+                 op_types_to_calibrate=[],
+                 augmented_model_path='augmented_model.onnx',
+                 method='percentile',
+                 num_quantized_bins=128,
+                 percentile=99.99):
```

Comment:
Will 1024 bins and 99.999% in general be better than 128 and 99.99% w.r.t accuracy?

Reply:
Pytorch quantization uses 2048 bins as default.
I tested 2048 bins and 99.999% which is better than 128 bins and 99.99%. Will change to these setting.

Reply:
Hi @chilo-ms ,  please check this issues, 
https://github.com/microsoft/onnxruntime/issues/9597